go/genkit/flow.go

```diff
@@ -695,48 +695,80 @@ func (f *Flow[In, Out, Stream]) run(ctx context.Context, input In, cb func(conte
 	return finishedOpResponse(state.Operation)
 }
 
-// StreamFlowValue is either a streamed value or a final output of a flow.
-type StreamFlowValue[Out, Stream any] struct {
-	Done   bool
-	Output Out    // valid if Done is true
-	Stream Stream // valid if Done is false
+// FlowIterator defines the interface for iterating over flow results.
+type FlowIterator[Out, Stream any] interface {
+	// Next returns the next streamed value and a boolean indicating whether the flow has completed.
+	Next() (Stream, bool)
+	// FinalOutput returns the final output of the flow if it has completed.
+	FinalOutput() (Out, error)
 }
 
-// Stream runs the flow on input and delivers both the streamed values and the final output.
-// It returns a function whose argument function (the "yield function") will be repeatedly
-// called with the results.
-//
-// If the yield function is passed a non-nil error, the flow has failed with that
-// error; the yield function will not be called again. An error is also passed if
-// the flow fails to complete (that is, it has an interrupt).
-// Genkit Go does not yet support interrupts.
-//
-// If the yield function's [StreamFlowValue] argument has Done == true, the value's
-// Output field contains the final output; the yield function will not be called
-// again.
-//
-// Otherwise the Stream field of the passed [StreamFlowValue] holds a streamed result.
-func (f *Flow[In, Out, Stream]) Stream(ctx context.Context, input In, opts ...FlowRunOption) func(func(*StreamFlowValue[Out, Stream], error) bool) {
```

Comment:
Ah, I actually wrote this with range-over-func in mind. Its return type is exactly an `iter.Seq2` (see https://pkg.go.dev/iter#Seq2). So think about keeping it. It seems weird now but won't in a few months.